vslib/src/SwitchStateBase.cpp

```diff
@@ -831,6 +831,19 @@ sai_status_t SwitchStateBase::set_switch_mac_address()
 
 sai_status_t SwitchStateBase::set_switch_default_attributes()
 {
+    // Fill this with supported SAI_OBJECT_TYPEs
+    int32_t supported_obj_list[] = {
+                                SAI_OBJECT_TYPE_NULL,
+                                SAI_OBJECT_TYPE_PORT,
+                                SAI_OBJECT_TYPE_LAG,
+                                SAI_OBJECT_TYPE_TAM,
+                                SAI_OBJECT_TYPE_TAM_COLLECTOR,
+                                SAI_OBJECT_TYPE_TAM_REPORT,
+                                SAI_OBJECT_TYPE_TAM_TRANSPORT,
+                                SAI_OBJECT_TYPE_TAM_TELEMETRY,
+                                SAI_OBJECT_TYPE_TAM_EVENT_THRESHOLD
+                              };
```

Comment:
actually all objects are supported, you can create any objects, but not all read only attributes will work, they need to be refreshed manually

Reply:
also NULL object should not be on that list ?

Reply:
use std::vector<sai_object_type_t> maybe ?

Reply:
also move that closes to the first use